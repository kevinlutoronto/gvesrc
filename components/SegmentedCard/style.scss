@import '../../styles/base';

.card {
  background: white;
  min-height: 20px;
  border: 1px solid $cBorder;
  border-radius: 3px;
  margin-bottom: 30px;
  display: flex;

  &.vertical {
    flex-direction: column;
  }
}

.first {
  padding: 40px 40px;
  flex-basis: 25%;
  background: rgba($cWhite, .7);
  border-right: 1px solid $cBorder;
  border-top-left-radius: 3px;
  border-bottom-left-radius: 3px;
}

.vertical {
  .first {
    border-right: 0;
    border-bottom: 1px solid $cBorder;
    flex: 1 0 default;
    padding-bottom: $space20;
  }
}

.second {
  padding: 40px;
  flex-grow: 1;
  border-top-right-radius: 3px;
  border-bottom-left-radius: 3px;
}

@media only screen and (min-device-width : 1px) and (max-device-width : 1024px) {
  .card {
    flex-direction: column;
  }
  .first {
    border-right: 0;
    border-bottom: 1px solid $cBorder;
    flex: 1 0 default;
    padding-bottom: $space20;
  }
}

