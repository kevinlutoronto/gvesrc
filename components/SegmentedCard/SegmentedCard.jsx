// @flow
import React from 'react';
import cx from 'classnames';
import style from './style.scss';
import { getDataAttrs } from '../../dataUtils';
import type { Data } from '../../types';

type Props = {|
  children: React$Node,
  style: Object,
  vertical: Boolean,
  data?: Data,
|};

class SegmentedCard extends React.Component<Props> {
  static defaultProps = {
    children: ['Side', 'Content'],
    style: {},
  };

  render() {
    const className = cx(style.card, {
      [style.vertical]: this.props.vertical,
    });
    const children = React.Children.toArray(this.props.children);

    return (
      <div {...getDataAttrs(this.props.data)} className={className} style={this.props.style}>
        <div className={style.first}>{children[0]}</div>
        <div className={style.second}>{children[1]}</div>
      </div>
    );
  }
}

SegmentedCard.displayName = 'Plasma@SegmentedCard';

export default SegmentedCard;
