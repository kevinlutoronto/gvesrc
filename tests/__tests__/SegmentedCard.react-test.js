/* eslint-disable */

import React from 'react';
import SegmentedCard from '../../components/SegmentedCard/SegmentedCard';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe('SegmentedCard Component', () => {
  test('render', () => {
    const component = renderer.create(<SegmentedCard />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
