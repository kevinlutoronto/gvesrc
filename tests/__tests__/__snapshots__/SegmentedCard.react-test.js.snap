// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedCard Component render 1`] = `
<div
  className="card"
  style={Object {}}
>
  <div
    className="first"
  >
    Side
  </div>
  <div
    className="second"
  >
    Content
  </div>
</div>
`;
